projects/wallet-service-reliable-integration/0001-design.md

```diff
@@ -0,0 +1,205 @@
+# Summary
+
+The goal of this document is to describe a new method of syncing with a Hathor node, using the reliable integrations feature and replace the current sync daemon on our production services
+
+# Motivation
+
+At present, our wallet-service sync daemon operates by utilizing the fullnode's WebSocket and HTTP APIs to maintain synchronization with the fullnode. With this approach, we have to manually handle reorganizations, making it not very maintenance friendly as we have to deal with all potential edge cases that are already handled in the fullnode, again in the daemon.
+  
+It was precisely with such applications in mind that we developed our '[Reliable Integrations](https://github.com/HathorNetwork/rfcs/blob/master/projects/reliable-integration/0001-high-level-design.md)' feature. The concept behind this feature is to implement a reliable, integrated event management system. Through this system, a series of events would be dispatched in the exact order of their occurrence, ensuring that if we follow this exact series of events, the same state will be obtained
+
+# Reference-level explanation
+
+In a nutshell, when the reliable integrations mechanism proposes a built-in event management system for applications interacting with a full-node. This system ensures events are sent in the order they occur, assigns unique IDs to each event, and offers both REST API and WebSocket access for event queries.
+
+Given the synchronous nature of the event processing, I propose we design the daemon as a finite state machine as it can simplify the design by making it explicit which actions are valid in each state and how the system transitions between states.
+
+Here is the proposed state diagram for handling fullnode's events:
+
+```mermaid
+stateDiagram-v2
+    [*] --> CONNECTING
+    CONNECTING --> CONNECTED: CONNECTED
```

Comment:
After entering the `CONNECTED` state, you must confirm the following:

1) You are connected to the same full node as before (i.e., same `peer-id`).
2) This full node is connected to the correct network (e.g., mainnet).

If any of those are not true, you should go to state `error`.

Reply:
Great! Thanks!